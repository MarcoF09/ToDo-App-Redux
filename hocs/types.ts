import React from 'react';

export type Hoc<TOuter = {}, TInner = TOuter> = (
  Component: React.ComponentType<TInner>
) => React.ComponentType<TOuter>;
export type ExtractTInner<T> = T extends Hoc<any, infer TInner>
  ? TInner
  : never;
