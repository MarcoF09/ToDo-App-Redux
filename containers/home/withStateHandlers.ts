import { Animated } from 'react-native';
import { withStateHandlers } from 'recompose';

const animateDeleteButton = (
  deleteButtonState: boolean,
  moveAnimation: Animated.ValueXY
): boolean => {
  if (!deleteButtonState) {
    Animated.spring(moveAnimation, {
      toValue: { x: 6, y: 30 }
    }).start();
    return true;
  }
  Animated.spring(moveAnimation, {
    toValue: { x: -100, y: 30 }
  }).start();
  return false;
};

export const withDeleteButtonHandler = withStateHandlers(
  {
    deleteButtonState: false,
    moveAnimation: new Animated.ValueXY({ x: -100, y: 30 })
  },
  {
    animateDeleteButton: state => () => ({
      deleteButtonState: animateDeleteButton(
        state.deleteButtonState,
        state.moveAnimation
      )
    })
  }
);
