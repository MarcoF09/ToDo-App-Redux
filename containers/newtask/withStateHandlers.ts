import { compose, withStateHandlers } from 'recompose';
import { Colors } from '../../colors/Colors';

const withTitleHandlers = withStateHandlers(
  { title: '' },
  {
    onChangeTitle: state => (text: string) => ({
      title: text
    })
  }
);

const withDescriptionHandlers = withStateHandlers(
  { description: '' },
  {
    onChangeDescription: state => (text: string) => ({
      description: text
    })
  }
);

const withTitleColorHandler = withStateHandlers(
  {
    firstInputColor: Colors.lightGrey
  },
  {
    onFocusTitle: state => () => ({
      firstInputColor: Colors.pink
    }),
    onBlurTitle: state => () => ({
      firstInputColor: Colors.lightGrey
    })
  }
);

const withDescriptionColorHandler = withStateHandlers(
  {
    secondInputColor: Colors.lightGrey
  },
  {
    onFocusDescription: state => () => ({
      secondInputColor: Colors.pink
    }),
    onBlurDescription: state => () => ({
      secondInputColor: Colors.lightGrey
    })
  }
);

export const stateEnhacer = compose(
  withTitleHandlers,
  withDescriptionHandlers,
  withTitleColorHandler,
  withDescriptionColorHandler
);
