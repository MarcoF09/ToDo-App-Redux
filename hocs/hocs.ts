import { noop } from 'lodash'
import React from 'react'
import { Hoc } from './types'

export function createHocTestComponent<TOuter = any, TInner = any>(
  hoc: Hoc<TOuter, TInner>,
  expectations: (props: TInner) => any = noop
): React.ComponentType<TOuter> {
  return hoc(props => {
    expectations(props)
    return null
  })
}
