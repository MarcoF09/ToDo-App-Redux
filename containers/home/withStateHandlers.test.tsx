import { shallow } from 'enzyme'
import { createHocTestComponent } from 'hocs/hocs'
import { defer } from 'lodash'
import React from 'react'
import { withDeleteButtonHandler } from './withStateHandlers'

describe('should animate the delete button', () => {
  const ComponentWithDeleteButtonHandler = createHocTestComponent(
    withDeleteButtonHandler
  )
  const wrapper = shallow(<ComponentWithDeleteButtonHandler />)
  it('brings the button', async () => {
    expect(wrapper.props().moveAnimation).toMatchSnapshot()
    expect(wrapper.props().deleteButtonState).toEqual(false)
    wrapper.props().animateDeleteButton(false, wrapper.props().moveAnimation)
    await Promise.resolve(defer)
    expect(wrapper.props().deleteButtonState).toEqual(true)
  })

  it('returns the button', async () => {
    wrapper.props().animateDeleteButton(true, wrapper.props().moveAnimation)
    await Promise.resolve(defer)
    expect(wrapper.props().deleteButtonState).toEqual(false)
  })
})
