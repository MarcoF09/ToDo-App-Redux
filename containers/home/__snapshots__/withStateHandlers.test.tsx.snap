// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should animate the delete button brings the button 1`] = `
Object {
  "x": -100,
  "y": 30,
}
`;
